src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/MultithreadedNodeMonitoringFailoverPluginTest.java

```diff
@@ -110,9 +114,12 @@ void test_1_executeSuccessful() throws ExecutionException, InterruptedException
     when(context.isNodeUnhealthy()).thenReturn(false);
     final int numConnections = 10;
     final int wantedNumberOfInvocations = NUM_PLUGINS_PER_SERVICE * numConnections;
-
     final List<NodeMonitoringFailoverPlugin> plugins = initPlugins(numConnections);
-    runExecuteAsync(numConnections, MONITOR_METHOD_NAME, plugins);
+    List<CompletableFuture<Object>> threads = runExecuteAsync(numConnections, plugins);
+
+    for (final CompletableFuture<Object> thread : threads) {
+      thread.get();
```

Comment:
What happens when you call thread.get() ?

Reply:
.get() blocks and waits for the thread to finish executing.